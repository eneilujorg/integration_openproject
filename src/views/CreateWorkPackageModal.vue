<template>
	<NcModal class="create-workpackage-modal"
		:show.sync="showModal"
		:can-close="true"
		:close-on-click-outside="false"
		:size="normal"
		@close="closeModal">
		<h2 class="create-workpackage-modal--title">
			{{ t('integration_openproject', 'Create and link a new work package') }}
		</h2>
		<LoadingIcon v-if="isLoading" class="loading-spinner" :size="40" />
		<iframe
			id="create-workpackage-iframe"
			width="600px"
			height="600px"
			:src="getIframeSource"
			@load="handleIframeLoad" />
	</NcModal>
</template>
<script>
import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
import { loadState } from '@nextcloud/initial-state'
import { STATE } from '../utils'
import LoadingIcon from 'vue-material-design-icons/Loading.vue'

export default {
	name: 'CreateWorkPackageModal',
	components: {
		NcModal,
		LoadingIcon,
	},
	props: {
		showModal: {
			type: Boolean,
			default: false,
		},
	},
	data: () => ({
		openprojectUrl: loadState('integration_openproject', 'openproject-url'),
		state: STATE.LOADING,
		show: false,
	}),
	computed: {
		getIframeSource() {
			const baseurl = this.openprojectUrl.replace(/\/+$/, '')
			return baseurl + '/work_packages/new?iframe=true'
		},
		isLoading() {
			return this.state === STATE.LOADING
		},
	},
	methods: {
		closeModal() {
			this.showModal = false
			this.$emit('closeCreateWorkPackageModal')
		},
		handleIframeLoad() {
			this.state = STATE.OK
			window.addEventListener('message', (event) => {
				if (event.origin !== this.openprojectUrl) return
				const eventData = event.data
				// send the data to the parent component to create link to the work package
				this.$emit('createWorkPackage', eventData)
			})
		},
	},
}
</script>
<style lang="scss">
.create-workpackage-modal {
	&--title {
		text-align: center;
		padding: 10px;
	}
}

.loading-spinner {
	display: flex;
	align-self: center;
	justify-self: center;
}
</style>
