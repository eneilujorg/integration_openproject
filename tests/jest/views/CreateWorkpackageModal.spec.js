/* jshint esversion: 8 */

import { createLocalVue, mount } from '@vue/test-utils'
import CreateWorkPackageModal from '../../../src/views/CreateWorkPackageModal.vue'
import * as initialState from '@nextcloud/initial-state'
import axios from '@nextcloud/axios'
import availableProjectsResponse from '../fixtures/openprojectAvailableProjectResponse.json'
import availableProjectsOption from '../fixtures/availableProjectOptions.json'
import workpackageFormValidationProjectSelected from '../fixtures/workpackageFormValidationProjectSelectedResponse.json'
import workpackageFormValidationTypeChanged from '../fixtures/workpackageFormValidationTypeChanged.json'
import availableProjectAssignees from '../fixtures/availableProjectAssigneesResponse.json'
import workpackageCreatedResponse from '../fixtures/workPackageSuccessfulCreationResponse.json'
import requiredTypeResponse from '../fixtures/formValidationResponseRequiredType.json'

const localVue = createLocalVue()

// eslint-disable-next-line no-import-assign,import/namespace
initialState.loadState = jest.fn(() => {
	return {
		openproject_instance_url: 'https://openproject.example.com',
	}
})

describe('CreateWorkPackageModal.vue', () => {
	const createWorkPackageSelector = '.create-workpackage-modal'
	const projectSelectSelector = '[data-test-id="available-projects"]'
	const statusSelectSelector = '[data-test-id="available-statuses"]'
	const typeSelectSelector = '[data-test-id="available-types"]'
	const assigneesSelectSelector = '[data-test-id="available-assignees"]'
	const projectInputField = '#createWorkPackageInput'
	const projectOptionsSelector = '[data-test-id="available-projects"] .vs__dropdown-menu .vs__dropdown-option'
	const typeOptionsSelector = '[data-test-id="available-types"] .vs__dropdown-menu .vs__dropdown-option'
	const typeInputFieldSelector = '#createWorkPackageTypeInput'
	const statusInputFieldSelector = '#createWorkPackageStatusInput'
	const assigneeInputFieldSelector = '#createWorkPackageAssigneeInput'
	const createWorkpackageButtonSelector = '.create-workpackage-form--button--create'
	const validationErrorSelector = '.validation-error'
	const validationErrorProjectSelector = '.multiple-error-project'
	const validationErrorSubjectSelector = '.multiple-error-subject'
	const validationErrorTypeSelector = '.type-error'
	let wrapper = null

	beforeEach(() => {
		jest.spyOn(document, 'getElementById').mockReturnValue({
			blur: jest.fn(), // Mock the blur function
		})
	})
	afterEach(async () => {
		wrapper.destroy()
		jest.restoreAllMocks()
	})

	describe('workpackage creation form', () => {
		it('should display available projects in the project dropdown', async () => {
			const axiosSpy = jest.spyOn(axios, 'get')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: availableProjectsResponse,
				}))
			wrapper = mountWrapper(true)
			expect(wrapper.find(createWorkPackageSelector).isVisible()).toBe(true)
			expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/projects')
			await wrapper.find(projectInputField).setValue(' ')
			expect(wrapper.find(projectSelectSelector)).toMatchSnapshot()
			axiosSpy.mockRestore()
			jest.clearAllMocks()
		})
		it('should set the available types, status and assignee when a project is selected', async () => {
			const formValidationBody = {
				body: {
					_links: {
						assignee: {
							href: null,
							title: null,
						},
						project: {
							href: '/api/v3/projects/2',
							title: 'Scrum project',
						},
						status: {
							href: '/api/v3/statuses/1',
							title: 'New',
						},
						type: {
							href: '/api/v3/types/1',
							title: 'Task',
						},
					},
					description: {
						format: 'markdown',
						html: '',
						raw: '',
					},
					subject: null,
				},
			}

			jest.spyOn(axios, 'get')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: availableProjectsResponse,
				}))
			const axiosSpy = jest.spyOn(axios, 'post')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: workpackageFormValidationProjectSelected,
				}))
			const assigneeAxiosSpy = jest.spyOn(axios, 'get')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: availableProjectAssignees,
				}))
			wrapper = mountWrapper(true, {
				noDropAvailableProjectDropDown: false,
			})
			wrapper.vm.mappedProjects = jest.fn(() => {
				return availableProjectsOption
			})
			const inputField = wrapper.find(projectInputField)
			await inputField.setValue('Scrum')
			await wrapper.find(projectOptionsSelector).trigger('click')
			await wrapper.vm.$nextTick()
			await wrapper.vm.$nextTick()
			expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/projects/2/work-packages/form', formValidationBody)
			expect(assigneeAxiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/projects/2/available-assignees')
			await wrapper.vm.$nextTick()
			await wrapper.find(typeInputFieldSelector).setValue(' ')
			await wrapper.vm.$nextTick()
			expect(wrapper.find(typeSelectSelector)).toMatchSnapshot()
			await wrapper.find(statusInputFieldSelector).setValue(' ')
			await wrapper.vm.$nextTick()
			expect(wrapper.find(statusSelectSelector)).toMatchSnapshot()
			await wrapper.find(assigneeInputFieldSelector).setValue(' ')
			await wrapper.vm.$nextTick()
			expect(wrapper.find(assigneesSelectSelector)).toMatchSnapshot()
			axiosSpy.mockRestore()
			assigneeAxiosSpy.mockRestore()
			jest.clearAllMocks()
		})

		it('should send the form validation request when type is changed', async () => {
			const formValidationBody = {
				body: {
					_links: {
						assignee: {
							href: null,
							title: null,
						},
						project: {
							href: '/api/v3/projects/2',
							title: 'Scrum project',
						},
						status: {
							href: '/api/v3/statuses/1',
							title: 'New',
						},
						type: {
							href: '/api/v3/types/2',
							title: 'Milestone',
						},
					},
					description: {
						format: 'markdown',
						html: '',
						raw: '',
					},
					subject: null,
				},
			}
			const allowedTypes = [
				{
					self: {
						href: '/api/v3/types/1',
						title: 'Task',
					},
					label: 'Task',
				},
				{
					self: {
						href: '/api/v3/types/2',
						title: 'Milestone',
					},
					label: 'Milestone',
				},
				{
					self: {
						href: '/api/v3/types/3',
						title: 'Phase',
					},
					label: 'Phase',
				},
				{
					self: {
						href: '/api/v3/types/5',
						title: 'Epic',
					},
					label: 'Epic',
				},
				{
					self: {
						href: '/api/v3/types/6',
						title: 'User story',
					},
					label: 'User story',
				},
				{
					self: {
						href: '/api/v3/types/7',
						title: 'Bug',
					},
					label: 'Bug',
				},
			]
			const availableStatusBefore = [
				{
					self: {
						href: '/api/v3/statuses/1',
						title: 'New',
					},
					label: 'New',
				},
				{
					self: {
						href: '/api/v3/statuses/7',
						title: 'In progress',
					},
					label: 'In progress',
				},
				{
					self: {
						href: '/api/v3/statuses/12',
						title: 'Closed',
					},
					label: 'Closed',
				},
				{
					self: {
						href: '/api/v3/statuses/13',
						title: 'On hold',
					},
					label: 'On hold',
				},
				{
					self: {
						href: '/api/v3/statuses/14',
						title: 'Rejected',
					},
					label: 'Rejected',
				},
			]
			const axiosSpy = jest.spyOn(axios, 'post')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: workpackageFormValidationTypeChanged,
				}))
			wrapper = mountWrapper(true, {
				allowedStatues: availableStatusBefore,
				allowedTypes,
				project: {
					self: {
						href: '/api/v3/projects/2',
						title: 'Scrum project',
					},
					label: '',
					children: [],
				},
				projectId: 2,
			})
			wrapper.vm.mappedProjects = jest.fn(() => {
				return availableProjectsOption
			})
			await wrapper.find(typeInputFieldSelector).setValue('Milest')
			await wrapper.find(typeOptionsSelector).trigger('click')
			await wrapper.vm.$nextTick()
			expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/projects/2/work-packages/form', formValidationBody)
			// one thing to note is the statues in snapshot should not match the statuses defined in variable availableStatusBefore
			await wrapper.find(statusInputFieldSelector).setValue(' ')
			expect(wrapper.find(statusSelectSelector)).toMatchSnapshot()
		})

		it.each([
			['should show error if the subject is empty',
				{
					project: {
						self: {
							href: '/api/v3/projects/2',
							title: 'Scrum project',
						},
						label: '',
						children: [],
					},
					projectId: 2,
					subject: null,
					data: "{\"_type\":\"Error\",\"errorIdentifier\":\"urn:openproject-org:api:v3:errors:PropertyConstraintViolation\",\"message\":\"Subject can't be blank.\",\"_embedded\":{\"details\":{\"attribute\":\"subject\"}}}",
					errorMessage: "Subject can't be blank.",
				}],
			['should show error if the project is empty',
				{
					project: {
						self: {
							href: null,
							title: null,
						},
						label: '',
						children: [],
					},
					projectId: null,
					subject: 'this is a workpackage',
					data: "{\"_type\":\"Error\",\"errorIdentifier\":\"urn:openproject-org:api:v3:errors:PropertyConstraintViolation\",\"message\":\"Project can't be blank.\",\"_embedded\":{\"details\":{\"attribute\":\"project\"}}}",
					errorMessage: "Project can't be blank.",
				}],
		])('%s', async (name, expectedErrorDetails) => {
			const createWorkpackageBody = {
				body: {
					_links: {
						type: {
							href: '/api/v3/types/1',
							title: 'Task',
						},
						status: {
							href: '/api/v3/statuses/1',
							title: 'New',
						},
						assignee: {
							href: null,
							title: null,
						},
						project: expectedErrorDetails.project.self,
					},
					subject: expectedErrorDetails.subject,
					description: {
						format: 'markdown',
						raw: '',
						html: '',
					},
				},
			}
			jest.spyOn(axios, 'get')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: availableProjectsResponse,
				}))
			const axiosSpy = jest.spyOn(axios, 'post')
				.mockImplementationOnce(() => Promise.resolve({
					status: 422,
					data: expectedErrorDetails.data,
				}))
			wrapper = mountWrapper(true, {
				project: expectedErrorDetails.project,
				projectId: expectedErrorDetails.projectId,
				subject: expectedErrorDetails.subject,
			})
			await wrapper.find(createWorkpackageButtonSelector).trigger('click')
			await wrapper.vm.$nextTick()
			expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/create/work-packages', createWorkpackageBody)
			const error = wrapper.find(validationErrorSelector)
			expect(error.isVisible()).toBe(true)
			expect(error.text()).toBe(expectedErrorDetails.errorMessage)
		})

		it('should show error if both project and subject are empty', async () => {
			const createWorkpackageBody = {
				body: {
					_links: {
						type: {
							href: '/api/v3/types/1',
							title: 'Task',
						},
						status: {
							href: '/api/v3/statuses/1',
							title: 'New',
						},
						assignee: {
							href: null,
							title: null,
						},
						project: {
							href: null,
							title: null,
						},
					},
					subject: null,
					description: {
						format: 'markdown',
						raw: '',
						html: '',
					},
				},
			}
			jest.spyOn(axios, 'get')
				.mockImplementationOnce(() => Promise.resolve({
					status: 200,
					data: availableProjectsResponse,
				}))
			const axiosSpy = jest.spyOn(axios, 'post')
				.mockImplementationOnce(() => Promise.resolve({
					status: 422,
					data: "{\"_type\":\"Error\",\"errorIdentifier\":\"urn:openproject-org:api:v3:errors:MultipleErrors\",\"message\":\"Multiple field constraints have been violated.\",\"_embedded\":{\"errors\":[{\"_type\":\"Error\",\"errorIdentifier\":\"urn:openproject-org:api:v3:errors:PropertyConstraintViolation\",\"message\":\"Subject can't be blank.\",\"_embedded\":{\"details\":{\"attribute\":\"subject\"}}},{\"_type\":\"Error\",\"errorIdentifier\":\"urn:openproject-org:api:v3:errors:PropertyConstraintViolation\",\"message\":\"Project can't be blank.\",\"_embedded\":{\"details\":{\"attribute\":\"project\"}}}]}}",
				}))
			wrapper = mountWrapper(true)
			await wrapper.find(createWorkpackageButtonSelector).trigger('click')
			await wrapper.vm.$nextTick()
			expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/create/work-packages', createWorkpackageBody)
			const projectError = wrapper.find(validationErrorProjectSelector)
			expect(projectError.isVisible()).toBe(true)
			expect(projectError.text()).toBe("Project can't be blank.")
			const subjectError = wrapper.find(validationErrorSubjectSelector)
			expect(subjectError.isVisible()).toBe(true)
			expect(subjectError.text()).toBe("Subject can't be blank.")
		})
	})

	it('should emit an event if work package creation is successful', async () => {
		const createWorkPackageBody = {
			body: {
				_links: {
					type: {
						href: '/api/v3/types/1',
						title: 'Task',
					},
					status: {
						href: '/api/v3/statuses/1',
						title: 'New',
					},
					assignee: {
						href: '/api/v3/users/15',
						title: 'Second Admin',
					},
					project: {
						href: '/api/v3/projects/2',
						title: 'Scrum project',
					},
				},
				subject: 'This is a workpackage',
				description: {
					format: 'markdown',
					raw: 'description for workpackage',
					html: '',
				},
			},
		}
		jest.spyOn(axios, 'get')
			.mockImplementationOnce(() => Promise.resolve({
				status: 200,
				data: availableProjectsResponse,
			}))
		const axiosSpy = jest.spyOn(axios, 'post')
			.mockImplementationOnce(() => Promise.resolve({
				status: 201,
				data: workpackageCreatedResponse,
			}))
		wrapper = mountWrapper(true, {
			project: {
				self: {
					href: '/api/v3/projects/2',
					title: 'Scrum project',
				},
				label: 'Scrum project',
				children: [],
			},
			assignee: {
				self: {
					href: '/api/v3/users/15',
					title: 'Second Admin',
				},
				label: 'Second Admin',
			},
			subject: 'This is a workpackage',
			description: {
				format: 'markdown',
				raw: 'description for workpackage',
				html: '',
			},
		})
		const emitSpy = jest.spyOn(wrapper.vm, '$emit')
		await wrapper.find(createWorkpackageButtonSelector).trigger('click')
		await wrapper.vm.$nextTick()
		expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/create/work-packages', createWorkPackageBody)
		expect(emitSpy).toHaveBeenCalledWith('create-work-package', {
			openProjectEventName: 'work_package_creation_success',
			openProjectEventPayload: workpackageCreatedResponse,
		})
	})

	it('should display error when there is a required custom field', async () => {
		const bodyFormValidation = {
			body: {
				_links: {
					type: {
						href: '/api/v3/types/9',
						title: 'Required CF',
					},
					status: {
						href: '/api/v3/statuses/1',
						title: 'New',
					},
					assignee: {
						href: null,
						title: null,
					},
					project: {
						href: '/api/v3/projects/4',
						title: '[dev] Large',
					},
				},
				subject: 'This is a workpackage',
				description: {
					format: 'markdown',
					raw: '',
					html: '',
				},
			},
		}
		const allowedTypes = [
			{
				self: {
					href: '/api/v3/types/9',
					title: 'Required CF',
				},
				label: 'Required CF',
			},
		]
		jest.spyOn(axios, 'get')
			.mockImplementationOnce(() => Promise.resolve({
				status: 200,
				data: availableProjectsResponse,
			}))
		const axiosSpy = jest.spyOn(axios, 'post')
			.mockImplementationOnce(() => Promise.resolve({
				status: 200,
				data: requiredTypeResponse,
			}))

		wrapper = mountWrapper(true, {
			project: {
				self: {
					href: '/api/v3/projects/4',
					title: '[dev] Large',
				},
				label: '[dev] Large',
				children: [],
			},
			type: {
				self: {
					href: '/api/v3/types/9',
					title: 'Required CF',
				},
				label: 'Required CF',
			},
			subject: 'This is a workpackage',
			allowedTypes,
			projectId: 2,
			openProjectUrl: 'https://openproject.example.com',
		})
		await wrapper.find(typeInputFieldSelector).setValue('Required')
		await wrapper.find(typeOptionsSelector).trigger('click')
		await wrapper.vm.$nextTick()
		expect(axiosSpy).toHaveBeenCalledWith('http://localhost/apps/integration_openproject/projects/2/work-packages/form', bodyFormValidation)
		const typeError = wrapper.find(validationErrorTypeSelector)
		expect(typeError.isVisible()).toBe(true)
		expect(typeError.text()).toBe('This type has mandatory fields which cannot be filled here. Please, create work packages of this type directly in {htmlLink}.')
	})

	it('should emit an event when the modal is closed', async () => {
		wrapper.vm.closeModal()
		expect(wrapper.emitted('close-create-work-package-modal')).toBeTruthy()
	})
})
function mountWrapper(showModal, data) {
	return mount(CreateWorkPackageModal, {
		localVue,
		mocks: {
			t: (app, msg) => msg,
			generateUrl() {
				return '/'
			},
		},
		data: () => ({
			...data,
		}),
		stubs: {
			NcModal: true,
		},
		propsData: {
			showModal,
		},
	})
}
