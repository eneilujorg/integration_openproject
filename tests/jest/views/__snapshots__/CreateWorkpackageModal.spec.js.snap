// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWorkPackageModal.vue workpackage creation form should display available projects in the project dropdown 1`] = `
<div dir="auto" class="v-select select create-workpackage-form--select vs--open vs--single vs--searching vs--searchable" data-test-id="available-projects">
  <div id="vs1__combobox" role="combobox" aria-expanded="true" aria-owns="vs1__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
    <div class="vs__selected-options"> <input placeholder="Select a project" id="createWorkPackageInput" aria-autocomplete="list" aria-labelledby="vs1__combobox" aria-controls="vs1__listbox" type="search" autocomplete="off" class="vs__search" aria-activedescendant="vs1__option-0"></div>
    <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear" style="display: none;"><span aria-hidden="true" role="img" class="material-design-icon close-icon" style="cursor: pointer;"><svg fill="var(--vs-controls-color)" width="20" height="20" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"><!----></path></svg></span></button> <span aria-hidden="true" role="presentation" class="material-design-icon chevron-down-icon vs__open-indicator"><svg fill="var(--vs-controls-color)" width="26" height="26" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"><!----></path></svg></span>
      <div class="vs__spinner" style="display: none;">Loading...</div>
    </div>
  </div>
  <transition-stub name="vs__fade">
    <ul id="vs1__listbox" role="listbox" tabindex="-1" class="vs__dropdown-menu">
      <li id="vs1__option-0" role="option" aria-selected="false" class="vs__dropdown-option vs__dropdown-option--highlight">
        <!----> <span>Demo project</span>
      </li>
      <li id="vs1__option-1" role="option" aria-selected="false" class="vs__dropdown-option">
        <!----> <span>[dev] Custom fields</span>
      </li>
      <li id="vs1__option-2" role="option" aria-selected="false" class="vs__dropdown-option">
        <!----> <span>[dev] Empty</span>
      </li>
      <li id="vs1__option-3" role="option" aria-selected="false" class="vs__dropdown-option">
        <!----> <span>[dev] Large</span>
      </li>
      <li id="vs1__option-4" role="option" aria-selected="false" class="vs__dropdown-option"><span style="padding-left: 2em;"></span> <span>[dev] Large child</span></li>
      <li id="vs1__option-5" role="option" aria-selected="false" class="vs__dropdown-option"><span style="padding-left: 4em;"></span> <span>[dev] Large Grand Child</span></li>
      <li id="vs1__option-6" role="option" aria-selected="false" class="vs__dropdown-option">
        <!----> <span>Scrum project</span>
      </li>
      <!---->
    </ul>
  </transition-stub>
</div>
`;

exports[`CreateWorkPackageModal.vue workpackage creation form should send the form validation request when type is changed 1`] = `
<div dir="auto" class="v-select select create-workpackage-form--select vs--open vs--single vs--searching vs--searchable" data-test-id="available-statuses">
  <div id="vs11__combobox" role="combobox" aria-expanded="true" aria-owns="vs11__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
    <div class="vs__selected-options"><span class="vs__selected"><span data-v-3daafbe0="" title="New" class="name-parts"><span data-v-3daafbe0="" class="name-parts__first">New</span>
      <!----></span>
      <!----></span> <input id="createWorkPackageStatusInput" aria-autocomplete="list" aria-labelledby="vs11__combobox" aria-controls="vs11__listbox" type="search" autocomplete="off" class="vs__search" aria-activedescendant="vs11__option-0">
    </div>
    <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear" style="display: none;"><span aria-hidden="true" role="img" class="material-design-icon close-icon" style="cursor: pointer;"><svg fill="var(--vs-controls-color)" width="20" height="20" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"><!----></path></svg></span></button> <span aria-hidden="true" role="presentation" class="material-design-icon chevron-down-icon vs__open-indicator"><svg fill="var(--vs-controls-color)" width="26" height="26" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"><!----></path></svg></span>
      <div class="vs__spinner" style="display: none;">Loading...</div>
    </div>
  </div>
  <transition-stub name="vs__fade">
    <ul id="vs11__listbox" role="listbox" tabindex="-1" class="vs__dropdown-menu">
      <li id="vs11__option-0" role="option" aria-selected="false" class="vs__dropdown-option vs__dropdown-option--highlight">
        To be scheduled
      </li>
      <li id="vs11__option-1" role="option" aria-selected="false" class="vs__dropdown-option">
        In progress
      </li>
      <li id="vs11__option-2" role="option" aria-selected="false" class="vs__dropdown-option">
        On hold
      </li>
      <!---->
    </ul>
  </transition-stub>
</div>
`;

exports[`CreateWorkPackageModal.vue workpackage creation form should set the available types, status and assignee when a project is selected 1`] = `
<div dir="auto" class="v-select select create-workpackage-form--select vs--open vs--single vs--searching vs--searchable" data-test-id="available-types">
  <div id="vs6__combobox" role="combobox" aria-expanded="true" aria-owns="vs6__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
    <div class="vs__selected-options"><span class="vs__selected"><span data-v-3daafbe0="" title="Task" class="name-parts"><span data-v-3daafbe0="" class="name-parts__first">Task</span>
      <!----></span>
      <!----></span> <input id="createWorkPackageTypeInput" aria-autocomplete="list" aria-labelledby="vs6__combobox" aria-controls="vs6__listbox" type="search" autocomplete="off" class="vs__search" aria-activedescendant="vs6__option-0">
    </div>
    <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear" style="display: none;"><span aria-hidden="true" role="img" class="material-design-icon close-icon" style="cursor: pointer;"><svg fill="var(--vs-controls-color)" width="20" height="20" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"><!----></path></svg></span></button> <span aria-hidden="true" role="presentation" class="material-design-icon chevron-down-icon vs__open-indicator"><svg fill="var(--vs-controls-color)" width="26" height="26" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"><!----></path></svg></span>
      <div class="vs__spinner" style="display: none;">Loading...</div>
    </div>
  </div>
  <transition-stub name="vs__fade">
    <ul id="vs6__listbox" role="listbox" tabindex="-1" class="vs__dropdown-menu">
      <li id="vs6__option-0" role="option" aria-selected="false" class="vs__dropdown-option vs__dropdown-option--highlight">
        User story
      </li>
      <!---->
    </ul>
  </transition-stub>
</div>
`;

exports[`CreateWorkPackageModal.vue workpackage creation form should set the available types, status and assignee when a project is selected 2`] = `
<div dir="auto" class="v-select select create-workpackage-form--select vs--open vs--single vs--searching vs--searchable" data-test-id="available-statuses">
  <div id="vs7__combobox" role="combobox" aria-expanded="true" aria-owns="vs7__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
    <div class="vs__selected-options"><span class="vs__selected"><span data-v-3daafbe0="" title="New" class="name-parts"><span data-v-3daafbe0="" class="name-parts__first">New</span>
      <!----></span>
      <!----></span> <input id="createWorkPackageStatusInput" aria-autocomplete="list" aria-labelledby="vs7__combobox" aria-controls="vs7__listbox" type="search" autocomplete="off" class="vs__search" aria-activedescendant="vs7__option-0">
    </div>
    <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear" style="display: none;"><span aria-hidden="true" role="img" class="material-design-icon close-icon" style="cursor: pointer;"><svg fill="var(--vs-controls-color)" width="20" height="20" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"><!----></path></svg></span></button> <span aria-hidden="true" role="presentation" class="material-design-icon chevron-down-icon vs__open-indicator"><svg fill="var(--vs-controls-color)" width="26" height="26" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"><!----></path></svg></span>
      <div class="vs__spinner" style="display: none;">Loading...</div>
    </div>
  </div>
  <transition-stub name="vs__fade">
    <ul id="vs7__listbox" role="listbox" tabindex="-1" class="vs__dropdown-menu">
      <li id="vs7__option-0" role="option" aria-selected="false" class="vs__dropdown-option vs__dropdown-option--highlight">
        In progress
      </li>
      <li id="vs7__option-1" role="option" aria-selected="false" class="vs__dropdown-option">
        On hold
      </li>
      <!---->
    </ul>
  </transition-stub>
</div>
`;

exports[`CreateWorkPackageModal.vue workpackage creation form should set the available types, status and assignee when a project is selected 3`] = `
<div dir="auto" class="v-select select create-workpackage-form--select vs--open vs--single vs--searching vs--searchable" data-test-id="available-assignees">
  <div id="vs8__combobox" role="combobox" aria-expanded="true" aria-owns="vs8__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
    <div class="vs__selected-options"> <input placeholder="Select a user or group" id="createWorkPackageAssigneeInput" aria-autocomplete="list" aria-labelledby="vs8__combobox" aria-controls="vs8__listbox" type="search" autocomplete="off" class="vs__search" aria-activedescendant="vs8__option-0"></div>
    <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear" style="display: none;"><span aria-hidden="true" role="img" class="material-design-icon close-icon" style="cursor: pointer;"><svg fill="var(--vs-controls-color)" width="20" height="20" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"><!----></path></svg></span></button> <span aria-hidden="true" role="presentation" class="material-design-icon chevron-down-icon vs__open-indicator"><svg fill="var(--vs-controls-color)" width="26" height="26" viewBox="0 0 24 24" class="material-design-icon__svg"><path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"><!----></path></svg></span>
      <div class="vs__spinner" style="display: none;">Loading...</div>
    </div>
  </div>
  <transition-stub name="vs__fade">
    <ul id="vs8__listbox" role="listbox" tabindex="-1" class="vs__dropdown-menu">
      <li id="vs8__option-0" role="option" aria-selected="false" class="vs__dropdown-option vs__dropdown-option--highlight">
        Second Admin
      </li>
      <li id="vs8__option-1" role="option" aria-selected="false" class="vs__dropdown-option">
        swikriti tripathi
      </li>
      <li id="vs8__option-2" role="option" aria-selected="false" class="vs__dropdown-option">
        OpenProject OpenProject
      </li>
      <li id="vs8__option-3" role="option" aria-selected="false" class="vs__dropdown-option">
        Admin de DEV user
      </li>
      <li id="vs8__option-4" role="option" aria-selected="false" class="vs__dropdown-option">
        OpenProject Admin
      </li>
      <!---->
    </ul>
  </transition-stub>
</div>
`;
